Documentation/Examples/RetriableSample.dib

```diff
@@ -0,0 +1,660 @@
+{
+
+  public RetryConfiguration retryConfiguration;
+  
+  public RetryOnRetryEnabledPolicy(RetryConfiguration retryConfiguration)
+  {
+      this.retryConfiguration = retryConfiguration;
+  }
+
+  public bool CanRetry(Exception exception)
+  {
+    return (!retryConfiguration.isRetryDisabled);
+  }
+}
+
+#!markdown
+
+Now, define an aggregate policy that combines `RetryOnRetryEnabledPolicy` & `RetryOnTooManyRequestsPolicy`
+
+#!csharp
+
+AggregatePolicy aggregatePolicy = new AggregatePolicy()
+{
+    Policies = 
+    {
+        new RetryOnTooManyRequestsPolicy(),
+        new RetryOnRetryEnabledPolicy(new RetryConfiguration())
+    }
+};
+
+#!markdown
+
+Now, try this aggregate policy with the `MockHttpService.MakeLotsOfRequests()` method.
+
+The default value of the `isRetryDisabled` property on `RetryConfiguration` is `false`, therefore the two conditions in `aggregatePolicy` will pass, and the method will be retried every time.
+
+#!csharp
+
+MockHttpService mockHttpService = new();
+Retriable.Retry(() => mockHttpService.MakeLotsOfRequests(), CancellationToken.None, strategy: new Count(10), policy: aggregatePolicy);
+
+#!markdown
+
+If we set  the `isRetryDisabled` property on `RetryConfiguration` to `false`, the condition in `RetryOnRetryEnabledPolicy` will not pass, and therefore `aggregatePolicy` will not trigger a retry.
+
+#!csharp
+
+var retryConfiguration = new RetryConfiguration();
+var retryOnRetryEnabledPolicy = new RetryOnRetryEnabledPolicy(retryConfiguration);
+
+AggregatePolicy aggregatePolicy = new AggregatePolicy()
+{
+    Policies = 
+    {
+        new RetryOnTooManyRequestsPolicy(),
+        retryOnRetryEnabledPolicy
+    }
+};
+
+// Change isRetryDisabled to true
+retryConfiguration.isRetryDisabled = true;
+
+MockHttpService mockHttpService = new();
+Retriable.Retry(() => mockHttpService.MakeLotsOfRequests(), CancellationToken.None, strategy: new Count(10), policy: aggregatePolicy);
+
+#!markdown
+
+If something similar were running in production, there may be a config change that prevents retries *whilst* a retry run is progress. Let's simulate that by starting a retry run (using the [`Linear`](#linear) strategy such that the delay between each retry is 2 seconds) asynchronously, and change the `isRetryDisabled` config property whilst it's running.
+
+#!csharp
+
+var retryConfiguration = new RetryConfiguration();
+var retryOnRetryEnabledPolicy = new RetryOnRetryEnabledPolicy(retryConfiguration);
+
+AggregatePolicy aggregatePolicy = new AggregatePolicy()
+{
+    Policies = 
+    {
+        new RetryOnTooManyRequestsPolicy(),
+        retryOnRetryEnabledPolicy
+    }
+};
+
+// Change isRetryDisabled to true
+//retryConfiguration.isRetryDisabled = true;
+
+MockHttpService mockHttpService = new();
+Action retryAction = () => Retriable.Retry(() => mockHttpService.MakeLotsOfRequests(), CancellationToken.None, strategy: new Linear(TimeSpan.FromSeconds(2), 10), policy: aggregatePolicy);
```

Comment:
This is great, but I'd consider maybe adding a `MakeLotsOfRequestsAsync()` method to your service and getting your Task from `Retriable.RetryAsync()` rather than starting the Task with a non-async retry inside it.

What you've done works fine, but what I've suggested is more "real world" and I have a slight concern that *not* using async retry in this case might make people think you can't (despite the async example above)

Reply:
I've updated the example like you've suggested.

Reply:
Most of the examples continue to be non-async, and I'm wondering if that's the right thing to do. In practice, I think we almost always use the async flavours of this. If you're retrying, it's probably because you're using some remote service that can fail.

Since the mocks you introduce are all stand-ins for remote requests, it seems like the right way to do this is to do async for almost everything, and possibly one non-async example just to show that that's supported. Right now it looks more like it's the other way around, which feels backwards to me.

Reply:
Updated to mostly use async examples.